tsc-lsif/src/lsif.ts

```diff
@@ -868,6 +918,32 @@ class MethodSymbolItem extends SymbolItem {
 		return referenceResult;
 	}
 
+	protected doResolveImplementationResult(emittingNode: ts.Node): ImplementationResult {

```

Comment:
Thinking about this again we now shift the time from computing the implementation result to querying it when we compute the result (e.g. we now have to find all the linked query results). I am actually not true anymore what is better. May be we also think about an inline result were we copy the range ids into every implementation result.

Reply:
I get the feeling that at one point or another, the results will have to be flattened.

The problem with an inline result is that in most cases the base class is emitted way before we process its implementations. I am trying to come up with a way of holding it back, but it might not be possible.